.NET/Microsoft.Recognizers.Text.Number/Japanese/Parsers/JapaneseNumberParser.cs

```diff
@@ -0,0 +1,344 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Recognizers.Text.Number.Japanese
+{
+    public class JapaneseNumberParser : BaseNumberParser
+    {
+        protected new readonly JapaneseNumberParserConfiguration Config;
+
+        public JapaneseNumberParser(JapaneseNumberParserConfiguration config) : base(config)
+        {
+            this.Config = config;
+        }
+
+        public override ParseResult Parse(ExtractResult extResultJpn)
+        {
+            string extra = null;
+            ParseResult ret = null;
+            extra = extResultJpn.Data as string;
+
+            var getExtResultJpn = new ExtractResult()
+            {
+                Start = extResultJpn.Start,
+                Length = extResultJpn.Length,
+                Data = extResultJpn.Data,
+                Text = extResultJpn.Text,
+                Type = extResultJpn.Type
+            };
+            
+            if (extra == null)
+            {
+                return null;
+            }
+
+            if (extra.Contains("Ordinal"))
+            {
+                ret = OrdParse(getExtResultJpn);
+            }
+            else if (extra.Contains("Per"))
+            {
+                ret = PerParse(getExtResultJpn);
+            }
+
+            if (ret != null)
+            {
+                ret.Text = extResultJpn.Text;
+            }
+
+            return ret;
+        }
+
+        // Replace full digtal numbers with half digtal numbers. "４" and "4" are both legal in Japanese, replace "４" with "4", then deal with "4"
+        private string ReplaceFullWithHalf(string text)
+        {
+            if (string.IsNullOrWhiteSpace(text))
+            {
+                return text;
+            }
+
+            var builder = new StringBuilder();
+            foreach (char c in text)
+            {
+                builder.Append(Config.FullToHalfMapJpn.ContainsKey(c) ? Config.FullToHalfMapJpn[c] : c);
+            }
+            return builder.ToString();
+        }
+
+        // Parse Japanese unit phrase. "万", "億",...
+        private string ReplaceUnit(string resultText)
+        {
+            foreach (var unit in Config.UnitMapJpn.Keys)
```

Comment:
UnitMapJpn, extResultJpn, etc.
Please use general names as much as possible to facilitate reading and reuse across languages.